src/azure-cli/azure/cli/command_modules/resource/_packing_engine.py

```diff
@@ -0,0 +1,188 @@
+    def __init__(self, root_template_directory):
+        self.RootTemplateDirectory = os.path.abspath(root_template_directory)
+        self.CurrentDirectory = os.path.abspath(root_template_directory)
+        self.Artifact = []
+
+
+# Packs the specified template and its referenced artifacts for use in a Template Spec.
+# :param template_file: The path to the template spec .json file.
+# :type name: str
+
+def pack(cmd, template_file):
+    root_template_file_path = os.path.abspath(template_file)
+    context = PackingContext(os.path.dirname(root_template_file_path))
+    template_content = read_file_content(template_file)
+    sanitized_template = _remove_comments_from_json(template_content)
+    template_json = json.loads(json.dumps(sanitized_template))
+    _pack_artifacts(cmd, root_template_file_path, context)
+    return PackagedTemplate(template_json, getattr(context, 'Artifact'))
+
+#  Recursively packs the specified template and its referenced artifacts and
+#  adds the artifacts to the current packing context.
+
+# :param template_abs_file_path: The path to the template spec .json file to pack.
+# :type template_abs_file_path : str
+# :param context : The packing context of the current packing operation
+# :type content : PackingContext
+# :param artifactableTemplateObj : The packagable template object
+# :type artifactableTemplateObj : JSON
+
+
+def _pack_artifacts(cmd, template_abs_file_path, context):
+    originalDirectory = getattr(context, 'CurrentDirectory')
+    try:
+        context.CurrentDirectory = os.path.dirname(template_abs_file_path)
+        template_content = read_file_content(template_abs_file_path)
+        artifactableTemplateObj = sanitized_template = _remove_comments_from_json(template_content)
+        template_json = json.loads(json.dumps(sanitized_template))
+        templateLinktoArtifactObjs = _get_template_links_to_artifacts(cmd, artifactableTemplateObj, includeNested=True)
+
+        for templateLinkObj in templateLinktoArtifactObjs:
+            relativePath = str(templateLinkObj['relativePath'])
+            if not relativePath:
+                continue
+            # This is a templateLink to a local template... Get the absolute path of the
+            # template based on its relative path from the current template directory and
+            # make sure it exists:
+
+            absoluteLocalPath = os.path.join(getattr(context, 'CurrentDirectory'), relativePath)
+            if not os.path.isfile(absoluteLocalPath):
+                raise CLIError('File ' + absoluteLocalPath + 'not found.')
+
+            # Let's make sure we're not referencing a file outside of our root directory
+            # hierarchy. We won't allow such references for security purposes:
+
+            if(not os.path.commonpath([getattr(context, 'RootTemplateDirectory')]) ==
+               os.path.commonpath([getattr(context, 'RootTemplateDirectory'), absoluteLocalPath])):
+                # TODO: Localize
```

Comment:
You can remove this TODO, this only applies to the PowerShell module which uses C# with allows you to create 
[resource files](https://docs.microsoft.com/en-us/dotnet/framework/resources/creating-resource-files-for-desktop-apps#resources-in-resx-files) to store strings that need to be localized. CLI doesn't have such files.